### ISSUE TITLE
Proposal Flow Components - Generics

### ISSUE THREAD

**Issue by elliotBraem** [NEAR-DevHub/neardevhub-bos#632]
## Overview 

This is a technical story to lay the foundation for the common components in the new proposal flow. This must be completed first in order to ensure consistency across pages and provide the common elements used across all pages.

We can condense the common components into the following:

- [x] Dropdown Select (Small)
- [ ] Dropdown Select (Large)
- [ ] Dropdown Button
- [ ] Input
- [ ] TextArea Editor
- [x] Markdown Editor (separate story)
- [ ] SidebarItem
- [ ] Avatar
- [ ] Checkbox
- [ ] Badge

![Image](https://github.com/NEAR-DevHub/neardevhub-bos/assets/16282460/5c0f056e-902c-4988-8297-045501d08e01)

Components can largely be influenced by the [Build DAO Component Library](https://www.nearbuilders.org/library) -- [index file on Github](https://github.com/NEARBuilders/gateway/blob/main/apps/builddao/widget/components.jsx)

Notice how components are functional components imported via VM.require, and only use a Widget when state is necessary (State is not necessary for the above list of common components)

**Comment by ori-near** [comment]
Hi @Megha-Dev-19 – is this ticket still in progress/relevant? Can you please update to reflect the latest status? Thank you.

**Comment by Megha-Dev-19** [comment]
Yeah sure let me attach it to the PR
